### PR TITLE
[Bug] Fixed CameraView crash on FlashMode Torch

### LINKED ISSUE #1314: [Bug] CameraView crashing when snapping picture with FlashMode set to Torch
### Description
When using Torch flash mode in a camera view, I'm unable to snap pictures without crashing.

### Steps to Reproduce

1. Create a CameraView with CaptureMode="Photo", CameraOptions="Back", FlashMode="{Binding CameraFlashMode}"
2. Toggle a button to switch from FlashMode="Off" to FlashMode="Torch"
3. Snap a picture

### Expected Behaviour
I expect to snap a photo having the flash always on (aka Torch mode)

### Actual Behavior

The flash mode changes correctly, but when I try to take a photo I get the following exception:

_Foundation.MonoTouchException: Objective-C exception thrown.  Name: NSInvalidArgumentException Reason: *** -[AVCapturePhotoOutput capturePhotoWithSettings:delegate:] flashMode must be set to a value present in the supportedFlashModes array_

### Basic Information

- Version with issue: 1.0.2, 1.2.0 pre-2
- Last known good version:  na
- IDE: VisualStudio 2019 for Mac
- Platform Target Frameworks: 
  - iOS:  14.4
- Nuget Packages: Xamarin.Forms 5.0.0.2012, Xamarin.Essentials 1.6.1, NETStandardLibrary 2.0.3, Network.Json 13.0.1
- Affected Devices: iPhone 6s

### PR DESCRIPTION
**Pull request by GUOLDEV**
Fixed CameraView crashing on FlashMode Torch

### Description of Change ###

The issue is that `photoSettings.FlashMode = (AVCaptureFlashMode)flashMode;` is converting `CameraFlashMode` to `AVCaptureFlashMode` and there is no option for `CameraFlashMode.Torch` in `AVCaptureFlashMode`.

### Bugs Fixed ###

- Fixes #1314 

### API Changes ###

<!-- List all API changes here (or just put None), example:

Added: 
 
- `string Class.Property { get; set; }`
- `void Class.Method();`

Changed:

 - `object Cell.OldPropertyName` => `object Cell.NewPropertyName`
 
If there is an entirely new API, then you can use a more verbose style:

```csharp
public static class NewClass {
    public static int SomeProperty { get; set; }
    public static void SomeMethod(string value);
}
``` -->

### Behavioral Changes ###

<!-- Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase. -->

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->
- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs [hunk 1]
```diff
@@ -239,6 +239,21 @@ public async Task TakePhoto()
...
+			switch (flashMode)
+			{
+				default:
+				case CameraFlashMode.Off:
+					return AVCaptureFlashMode.Off;
+				case CameraFlashMode.On:
+				case CameraFlashMode.Torch:
+					return AVCaptureFlashMode.On;
+				case CameraFlashMode.Auto:
+					return AVCaptureFlashMode.Auto;
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs:253]: Just changing the `default` order to match with the rest of our code
```suggestion
				case CameraFlashMode.On:
				case CameraFlashMode.Torch:
					return AVCaptureFlashMode.On;
				case CameraFlashMode.Auto:
					return AVCaptureFlashMode.Auto;
				case CameraFlashMode.Off:
				default:
					return AVCaptureFlashMode.Off;
```

Comment (GUOLDEV) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs]: @pictos done 👍

**Review by pictos — APPROVED** [review]
Code looks good to me, but I can't test it locally, because I don't a physical device to test it.

**Comment by GUOLDEV** [comment]
Thanks @pictos 👍  @jfversluis could you please have a look when you have a chance?

**Review by AndreiMisiukevich — APPROVED** [review]
LGTM

**Comment by GUOLDEV** [comment]
Thanks @AndreiMisiukevich 👍

**Pull request merged**